core/chainio/avs_writer.go

```diff
@@ -91,3 +103,86 @@ func (w *AvsWriter) SendAggregatedResponse(batchMerkleRoot [32]byte, senderAddre
 
 	return &txHash, nil
 }
+
+func (w *AvsWriter) checkRespondToTaskFeeLimit(tx *types.Transaction, txOpts bind.TransactOpts, batchIdentifierHash [32]byte, senderAddress [20]byte) error {
+	aggregatorAddress := txOpts.From
+	simulatedCost := new(big.Int).Mul(new(big.Int).SetUint64(tx.Gas()), tx.GasPrice())
+	w.logger.Info("Simulated cost", "cost", simulatedCost)
+
+	// Get RespondToTaskFeeLimit
+	batchState, err := w.AvsContractBindings.ServiceManager.BatchesState(&bind.CallOpts{}, batchIdentifierHash)
+	if err != nil {
+		// Retry with fallback
+		batchState, err = w.AvsContractBindings.ServiceManagerFallback.BatchesState(&bind.CallOpts{}, batchIdentifierHash)
+		if err != nil {
+			// Fallback also failed, proceed with simulated cost checks
+			w.logger.Error("Failed to get batch state", "error", err)
+			w.logger.Info("Proceeding with simulated cost checks")
+			if err := w.compareAggregatorBalance(simulatedCost, aggregatorAddress); err != nil {
+				return err
+			}
+			if err := w.compareBatcherBalance(simulatedCost, senderAddress); err != nil {
+				return err
+			}
```

Comment:
smol nitpick, this sequence could be moved to one function since we are doing it twice

Reply:
Sure why not, done